include/vara/Feature/ConstraintParser.h

```diff
@@ -257,6 +260,31 @@ class ConstraintLexer {
 //                               ConstraintParser
 //===----------------------------------------------------------------------===//
 
+/// Parse 64-bit integer in decimal or scientific notation.
+static int64_t parseInteger(llvm::StringRef Str) {
+  int64_t Integer;
+  if (!Str.getAsInteger(10, Integer)) {
+    return Integer;
+  }
+
+  if (Str.contains_lower('e')) {
+    // If we encounter scientific notation we try to parse the number as double.
+    double Double = 0;
+    if (!Str.getAsDouble(Double)) {
+      return round(Double);
```

Comment:
We need to add a check here, to war should a user pass an value that's to large.